### DISCUSSION TITLE
Connecting to rest-server

### DISCUSSION THREAD

**Discussion by dimejo** [rustic-rs/rustic#27] [Q&A] [answered]
I'm unable to connect to a rest-server with rustic. The command I have used:
```
rustic -r rest:http://127.0.0.1:8000/reponame -p repo.pass snapshots
Error: No config file found. Is there a repo?
```
The same command does work with restic. Any ideas?

**Comment by aawsome** [comment]
Uh, rustic seems to omit the path of the rest url, in your case `reponame`. So, this is clearly a bug in the REST-backend. I'll look after it.

The only workaround so far is to only use the rest-server for one repo and use an url without path.

**Comment by aawsome** [comment]
Actually, `rustic -r rest:http://127.0.0.1:8000/reponame/ -p repo.pass snapshots` (with a `/` at the end) should work.
Seems like reqwest uses `http://127.0.0.1:8000/` as base url if `reponame` is not terminated by an slash.

IMO this is still a bug, but as workaround adding the `/` should be ok.

**Comment by dimejo** [comment]
> IMO this is still a bug, but as workaround adding the / should be ok.

Unfortunately this yields the same error message for me. 🤷

**Reply by aawsome** [reply]
Ah, thanks for the feedback. I only tested with a working repo under `/` and checked some of the URL the rest-server got. So there is also a bug when checking for the config file. I added this to #29. Can you try if the PR works for you?

**Comment by dimejo** [comment, marked as answer]
#29 works for me. Thanks for the fast fix! 👍

**Reply by aawsome** [reply]
#29 is now merged into main.
